app/src/ui/add-repository/sanitized-repository-name.ts

```diff
@@ -1,4 +1,12 @@
 /** Sanitize a proposed repository name by replacing illegal characters. */
 export function sanitizedRepositoryName(name: string): string {
-  return name.replace(/[^\w.-]/g, '-')
+  const ranges = [
+    '\ud83c[\udf00-\udfff]', // U+1F300 to U+1F3FF
+    '\ud83d[\udc00-\ude4f]', // U+1F400 to U+1F64F
+    '\ud83d[\ude80-\udeff]', // U+1F680 to U+1F6FF
+  ]
+
+  return name
+    .replace(new RegExp(ranges.join('|'), 'g'), '-')
```

Comment:
How do you feel about moving `ranges` and the creation of the `RegExp` object to outside of this function?

```ts
const ranges = []
const emojiRE = new RegExp(ranges.join('|'), 'g')

export function sanitizedRepositoryName(name: string): string {
}
```

Doing it this way means we won't create the `RegExp` object every time we run this function.

Reply:
Done